### PR TITLE
README updates

### PR DESCRIPTION
**Pull request by Be-ing**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -25,7 +25,10 @@ The [CXX-Qt Book](https://kdab.github.io/cxx-qt/book/getting-started/index.html)
 step-by-step and documents CXX-Qt's features for the latest release. The [examples folder](./examples) contains
 demonstrations of using threading, QQmlExtensionPlugin, and various other features.
 
-CXX-Qt is in early development and the API changes frequently. For the latest documentation, [install mdBook](https://rust-lang.github.io/mdBook/guide/installation.html)
+CXX-Qt is tested on CI on Linux, Windows, and macOS (all on x86). It should work on other platforms that Qt and
```
Comment (ahayzen-kdab) [inline README.md:28]: "tested" on Windows is maybe a little strong ... it builds on Windows :-) the examples do run, however they segfault on exit IIRC.

File: README.md [hunk 2]
```diff
@@ -56,13 +59,15 @@ Ensure that you have the following installed
...
-  * [Rust](https://www.rust-lang.org/)
-  * Linux 64-bit x86 - currently we only support Linux, but we plan on adding arm 64-bit, macOS, and Windows support in the future
+  * [Qt 5 and/or Qt 6](https://www.qt.io/)
+  * [Rust toolchain](https://www.rust-lang.org/)
+  * [mold](https://github.com/rui314/mold) or [lld](https://lld.llvm.org/) for Linux (lld is included in the XCode toolchain on macOS)
 
-### Compiling
-In a CXX-Qt project, the build system is based on CMake, which uses Cargo under the hood.
-Therefore, unlike a typical Rust project, CMake must be used to build CXX-Qt.
+This repository's build system uses CMake, which calls Cargo under the hood to build all the
```
Comment (ahayzen-kdab) [inline README.md:66]: Are the individual examples still buildable with CMake by themselves ? and should we mention this ?

Eg can you enter the `qml_minimal` folder and do `mkdir build; cd build; cmake ..; make; ./qml_minimal` ?

Comment (Be-ing) [reply README.md:66]: They should be, but I don't think we need to mention this. There's already a lot of text in this section.

File: README.md (hunk 1 above)
Comment (Be-ing) [reply README.md:28]: I just tested on Windows and they do not segfault on exit. However, some more instructions are needed to set all the environment variables Windows needs...

Comment (Be-ing) [reply README.md:28]: I think this language is fine for now, but we do need to automate/document getting set up on Windows later.

File: README.md [hunk 3]
```diff
@@ -25,7 +25,10 @@ The [CXX-Qt Book](https://kdab.github.io/cxx-qt/book/getting-started/index.html)
 step-by-step and documents CXX-Qt's features for the latest release. The [examples folder](./examples) contains
 demonstrations of using threading, QQmlExtensionPlugin, and various other features.
 
-CXX-Qt is in early development and the API changes frequently. For the latest documentation, [install mdBook](https://rust-lang.github.io/mdBook/guide/installation.html)
+CXX-Qt is tested on CI on Linux, Windows, and macOS (all on x86). It should work on other platforms that Qt and
+Rust both support, however, these are untested.
```
[the remark this answers is not in the document]
Comment (Be-ing) [reply README.md:29]: Yeah, but we'd need ARM hardware for that. You can use qemu on Github Actions to emulate ARM, but it's extremely slow.

**Review by ahayzen-kdab — APPROVED** [review]
LGTM :-)

**Pull request merged**